src/index.ts

```diff
@@ -1,44 +1,223 @@
+      repeat: 11, //NUMBER OF STARS TO CREATE
+      setXY: { x: 12, y: 0, stepX: 70 } //XY ORIGIN
+  });
+
+  //SET RANDOM BOUNCE TO ADD VARIETY TO COLLECTABLES
+  stars.children.iterate(function (child) {
+      child.setBounceY(Phaser.Math.FloatBetween(0.4, 0.8));
+  });
+
+  //OVERLAP ALL
+  this.physics.add.overlap(player, stars, collectStar, null, this);
+
+  //COLLECTABLES COLLISION
+  this.physics.add.collider(stars, platforms);
+
+  //PLATFORM + PLAYER COLLISION
+  this.physics.add.collider(player, platforms);
+
+  //PLATFORM + BOMB COLLISION (BOUNCE)
+  this.physics.add.collider(bombs, platforms);
+
+  //PLAYER + BOMB COLLISION (DIE)
+  this.physics.add.collider(player, bombs, hitBomb, null);
+
+  //ADD CONTROLLER
+  cursors = this.input.keyboard.createCursorKeys();
+
+  //DISPLAY TEXT
+  scoreText = this.add.text(16, 16, 'score: 0', { fontSize: '32px', fill: '#000' });
+
+  //PLAYER ANIMATIONS
+  //LEFT MOVEMENT ANIMATION
+  this.anims.create({
+      key: 'left',
+      frames: this.anims.generateFrameNumbers('dude', { start: 0, end: 3 }),
+      frameRate: 10,
+      repeat: -1
   });
+
+  //TURN ANIMATION (LOOKING FORWARD)
+  this.anims.create({
+      key: 'turn',
+      frames: [ { key: 'dude', frame: 4 } ],
+      frameRate: 20
+  });
+
+  //RIGHT MOVEMENT ANIMATION
+  this.anims.create({
+      key: 'right',
+      frames: this.anims.generateFrameNumbers('dude', { start: 5, end: 8 }),
+      frameRate: 10,
+      repeat: -1
+  });
+
+}
+
+function update()
+{
+  //MOVE PLAYER & ANIMATE
+  if (cursors.left.isDown) //IS LEFT ARROW DOWN?
+  {
+      player.setVelocityX(-160); //MOVE LEFT
+      player.anims.play('left', true); //& PLAY LEFT ANIMATION
+  }
+  else if (cursors.right.isDown) //IS RIGHT ARROW DOWN?
+  {
+      player.setVelocityX(160); //MOVE RIGHT
+      player.anims.play('right', true); //& PLAY RIGHT ANIMATION
+  }
+  else //IF NO KEYS ARE PRESSED
+  {
+      player.setVelocityX(0); //NO MOVEMENT
+      player.anims.play('turn'); //PLAY TURN ANIMATION
+  }
+
+
+  //JUMP!
+  if (cursors.up.isDown && player.body.touching.down)
+  {
+      player.setVelocityY(-330); //ADD UPWARD VELOCITY
+  }
+
+}
+
+//DELETE STARS WHEN COLLECTED
+function collectStar (player, star)
+{
+    //DISABLE STAR
+    star.disableBody(true, true);
+
+    //ADD POINTS TO SCORE
+    score += 10;
+
+    //UPDATE SCORE
+    scoreText.setText('Score: ' + score);
+
+    if (stars.countActive(true) === 0)
+    {
+        stars.children.iterate(function (child) {
+
+            child.enableBody(true, child.x, 0, true, true);
+
+        });
+    }
+
+    drawBomb();
+}
+
+//PLAYER DEATH FROM BOMB
+function hitBomb (player, bomb)
+{
+    //MAKE PLAYER RED --- FOR SOME REASON THIS DOESNT WORK? IT WORKS IF WE REMOVE PAUSE()
+    //MAYBE PHYSICS ARE UPDATED MORE FREQUENTLY? MAYBE A CUSTOM GAME LOOP CAN FIX?
+    player.setTint(0xff0000);
+
+    //SET TO FORWARD FACING FRAME
+    player.anims.play('turn');
+
+    //STOP GAME
+    this.physics.pause();
+
```

Comment:
The issue with [`setTint`](https://photonstorm.github.io/phaser3-docs/Phaser.GameObjects.Sprite.html#setTint) was actually not with `setTint`, it was with how we did not set the collider's `callbackContext` further up when we defined the `collideCallback` to be `hitBomb`. I commented on that line as to how to fix.

It is not completely obvious at first glance, but because we didn't set the context, when you use `this`, the function doesn't have the correct context to grab the physics object. What was happening was the `this.physics` in `this.physics.pause();` was undefined. If you noticed the console in your browser it was logging an error saying something like `this.physics` is undefined. The tint was getting set correctly, the method was just erroring later on and never completing.

See: https://photonstorm.github.io/phaser3-docs/Phaser.Physics.Arcade.Collider.html When the code goes to `this.physics.pause()`